starcheck/src/lib/Ska/Starcheck/Obsid.pm

```diff
@@ -2656,14 +2634,235 @@ sub set_ccd_temps{
+                             include_ids=kw['acqs'], include_halfws=kw['halfwidths'])
+
+    # Assign the proseco probabilities back into an array.
+    p_acqs = [float(acq_cat['p_acq'][acq_cat['id'] == acq][0]) for acq in kw['acqs']]
+
+    return p_acqs, float(acq_cat.get_log_p_2_or_fewer()), float(np.sum(p_acqs))
+};
+
+
+###################################################################################
+sub proseco_args{
+###################################################################################
+# Build a hash that corresponds to reasonable arguments to use to call proseco get_acq_catalog
+# to calculate marginalized acquisition probabilities for a star catalog.
+# This routine also saves the guides and fids into lists, but those are not used
+# by get_acq_catalog.
+# If an observation does not have a target quaternion or a starcat, it is skipped and
+# an empty hash is returned with no warning.
+    my $self = shift;
+    my %proseco_args;
+    # For the target quaternion, use the -1 to get the last quaternion (there could be more than
+    # one for a segmented maneuver).
+    my $targ_cmd = find_command($self, "MP_TARGQUAT", -1);
+    my $cat_cmd = find_command($self, "MP_STARCAT");
+    # For observations without a target attitude, catalog, or defined obsid return an empty hash
+    if ((not $targ_cmd) or (not $cat_cmd) or ($self->{obsid} =~ /NONE(\d+)/)){
+        return \%proseco_args;
+    }
+    # Use a default SI (which should only be used for ERs and should have no effect without fid lights)
+    my $si = 'ACIS-S';
+    my $offset = 0;
+    if ($self->{obsid} < 38000){
+        $si = $self->{SI};
+        $offset = $self->{SIM_OFFSET_Z};
+    }
+
+    my @acq_ids;
+    my @acq_indexes;
+    my @gui_ids;
+    my @fid_ids;
+    my @halfwidths;
+    # Loop over the star catalog and assign the acqs, guide stars, and fids to arrays.
+  IDX:
+    foreach my $i (1..16) {
+        (my $sid  = $cat_cmd->{"GS_ID$i"}) =~ s/[\s\*]//g;
+        # If there is no star there is nothing for proseco probs to do so skip it.
+        # But warn if it was a thing that should have had an id (BOT/ACQ/GUI).
+        if ($sid eq '---'){
+            if ($cat_cmd->{"TYPE$i"} =~ /BOT|ACQ|GUI/){
+                push @{$self->{warn}}, sprintf(
+                    ">> WARNING: [%2d] Could not calculate acq prob for star with no id.", $i);
+            }
+            next IDX;
+        }
+        $sid = int($sid);
+        # While assigning ACQ stars into a list, clip to allowed ranges and warn as needed.
+        if ($cat_cmd->{"TYPE$i"} =~ /BOT|ACQ/){
+            push @acq_ids, $sid;;
+            my $hw = $cat_cmd->{"HALFW$i"};
+            if (($hw > 180) or ($hw < 60)){
+                push @{$self->{yellow_warn}}, sprintf(
+                    ">> WARNING: [%2d] Halfwidth %d outside range 60 to 180. Clipped for probs.",
+                    $i, $hw);
+                # Clip hw if outside the range 60 to 180 for probabilities
```

Comment:
Can you restate?  I'm lost.  As far as I can tell, proseco UserWarns on all the clips it does, and our previously discussion was that we didn't want those as UserWarnings at console.

Reply:
Where do you see that?  I actually tested this prior to making the comment and didn't see warnings.
```
In [7]: aca = get_aca_catalog(21277, include_ids_acq=[1152523976], include_halfws_acq=[240])

In [8]: aca
```
Here's the line in question, which just says to extrapolate:

https://github.com/sot/proseco/blob/24ad845442861440122af028ebffbcb4ca01dfcb/proseco/acq.py#L473

Reply:
Gotcha.  I didn't realize proseco was interpolating to effect a clip before passing to the grid acq model.  Doesn't matter *that* much because I think we should have visibility in starcheck if a box size outside the range used by the model or proseco is picked.  Not sure about the best way to implement in that case.

Specifically, I'm happier to also get rid of the "search box smaller that slew error" warning if I know that there is something that will caution it if we have a zero halfwidth acq box.

Reply:
Oh wait, the proseco one is extrapolating?  I look at this some more.

Reply:
"Extrapolating" in this case only means that it doesn't throw an exception if you give it a value outside the range. The `kind='nearest'` arg is key here, which is nearest-neighbor interpolation, or the relevant edge value for x-values outside the bounds.  It works, I tested it.

Reply:
The warning on search box sizes outside of the proseco range is good, I was not implying to drop that.

Reply:
And now I realize where you were seeing the UserWarning, which is that starcheck 12.1 calls the chandra_aca star probabilities model directly and can feed out-of-bounds halfwidths (e.g. JAN0419A).  But starcheck 13.0+ will use proseco, which does the clipping without issuing a warning.